File: backend/mock_data.py
# Python mock data for UCEK Unstop Igniters Placement Platform

INITIAL_QUESTIONS = []

INITIAL_MOCK_TESTS = [
  {
    "id": "tcs_ninja_drive",
    "title": "TCS Ninja & Digital National Drive Mock 2026",
    "category": "Company Drive",
    "companyTag": "TCS",
    "company_tag": "TCS",
    "durationMinutes": 30,
    "duration_mins": 30,
    "passPercentage": 70,
    "pass_percentage": 70,
    "totalQuestions": 5,
    "questions": [
      {
        "id": "q1",
        "title": "A train 150m long passes a telegraph post in 12 seconds. What is the speed of the train in km/hr?",
        "options": ["45 km/hr", "50 km/hr", "36 km/hr", "40 km/hr"],
        "correctOption": 0
      },
      {
        "id": "q2",
        "title": "Which data structure is primarily used to implement Recursion in programming languages?",
        "options": ["Queue", "Call Stack", "Min-Heap", "Graph"],
        "correctOption": 1
      },
      {
        "id": "q3",
        "title": "Select the antonym for the word 'BENEVOLENT':",
        "options": ["Malevolent", "Generous", "Altruistic", "Friendly"],
        "correctOption": 0
      }
    ]
  },
  {
    "id": "infosys_nqt",
    "title": "Infosys Specialist Programmer Diagnostic Exam",
    "category": "Company Drive",
    "companyTag": "Infosys",
    "company_tag": "Infosys",
    "durationMinutes": 25,
    "duration_mins": 25,
    "passPercentage": 75,
    "pass_percentage": 75,
    "totalQuestions": 4,
    "questions": [
      {
        "id": "iq1",
        "title": "Find the next number in the series: 3, 5, 9, 17, 33, ?",
        "options": ["65", "49", "64", "55"],
        "correctOption": 0
      },
      {
        "id": "iq2",
        "title": "In Time Complexity analysis, what does Big-Omega (Ω) notation denote?",
        "options": ["Best Case", "Worst Case", "Average Case", "Upper Bound"],
        "correctOption": 0
      }
    ]
  }
]

INITIAL_INTERVIEW_QUESTIONS = [
  {
    "id": "hr_1",
    "question": "Tell me about yourself and why you are interested in joining our organization as a Campus Recruit.",
    "questionText": "Tell me about yourself and why you are interested in joining our organization as a Campus Recruit.",
    "category": "HR & Behavioral",
    "companyTag": "TCS",
    "company_tag": "TCS",
    "isFeatured": True
  },
  {
    "id": "hr_2",
    "question": "Describe a challenging situation during your final year project where team conflict arose. How did you resolve it?",
    "questionText": "Describe a challenging situation during your final year project where team conflict arose. How did you resolve it?",
    "category": "HR & Behavioral",
    "companyTag": "TCS",
    "company_tag": "TCS",
    "isFeatured": True
  },
  {
    "id": "inf_1",
    "question": "Why do you want to join Infosys as a Systems Engineer / Specialist Programmer, and where do you see yourself in 3 years?",
    "questionText": "Why do you want to join Infosys as a Systems Engineer / Specialist Programmer, and where do you see yourself in 3 years?",
    "category": "HR & Behavioral",
    "companyTag": "Infosys",
    "company_tag": "Infosys",
    "isFeatured": True
  },
  {
    "id": "inf_2",
    "question": "What is the difference between Process and Thread? Explain how Context Switching works.",
    "questionText": "What is the difference between Process and Thread? Explain how Context Switching works.",
    "category": "Technical",
    "companyTag": "Infosys",
    "company_tag": "Infosys",
    "isFeatured": True
  },
  {
    "id": "wip_1",
    "question": "How do you prioritize multiple tasks when working under tight deadlines in a multi-client project team?",
    "questionText": "How do you prioritize multiple tasks when working under tight deadlines in a multi-client project team?",
    "category": "Situational",
    "companyTag": "Wipro",
    "company_tag": "Wipro",
    "isFeatured": True
  },
  {
    "id": "wip_2",
    "question": "Describe a project where you had to learn a new technology or programming language quickly.",
    "questionText": "Describe a project where you had to learn a new technology or programming language quickly.",
    "category": "HR & Behavioral",
    "companyTag": "Wipro",
    "company_tag": "Wipro",
    "isFeatured": True
  },
  {
    "id": "acc_1",
    "question": "Accenture focuses heavily on innovation and emerging tech. Share an example of how you used creative problem-solving in a project.",
    "questionText": "Accenture focuses heavily on innovation and emerging tech. Share an example of how you used creative problem-solving in a project.",
    "category": "HR & Behavioral",
    "companyTag": "Accenture",
    "company_tag": "Accenture",
    "isFeatured": True
  },
  {
    "id": "acc_2",
    "question": "How do you handle constructive criticism or feedback from a senior developer or team mentor?",
    "questionText": "How do you handle constructive criticism or feedback from a senior developer or team mentor?",
    "category": "HR & Behavioral",
    "companyTag": "Accenture",
    "company_tag": "Accenture",
    "isFeatured": True
  },
  {
    "id": "gen_1",
    "question": "What are your key strengths and what is one technical area you are actively working to improve?",
    "questionText": "What are your key strengths and what is one technical area you are actively working to improve?",
    "category": "HR & Behavioral",
    "companyTag": "General HR",
    "company_tag": "General HR",
    "isFeatured": True
  },
  {
    "id": "gen_2",
    "question": "If you are given two high-priority tasks with tight deadlines simultaneously by two different project leads, how will you manage?",
    "questionText": "If you are given two high-priority tasks with tight deadlines simultaneously by two different project leads, how will you manage?",
    "category": "Situational",
    "companyTag": "General HR",
    "company_tag": "General HR",
    "isFeatured": True
  }
]

DEFAULT_ROADMAPS = {
  "Software Engineering": [
    {
      "id": "m1",
      "title": "Core CS Fundamentals",
      "description": "Data Structures, Algorithms, OOPs, OS & DBMS for TCS/Infosys drives",
      "milestones": [
        {"id": "ms1", "title": "Arrays, Strings & Two Pointers", "completed": True},
        {"id": "ms2", "title": "Stacks, Queues & HashMaps", "completed": True},
        {"id": "ms3", "title": "SQL Queries & Indexing", "completed": False},
        {"id": "ms4", "title": "Process Scheduling & Deadlocks", "completed": False}
      ]
    },
    {
      "id": "m2",
      "title": "Full Stack Web Development",
      "description": "React.js, Node.js/Python, REST APIs & Cloud Deployment",
      "milestones": [
        {"id": "ms5", "title": "HTML5, CSS3, Tailwind & Modern ES6+", "completed": True},
        {"id": "ms6", "title": "React Hooks & State Management", "completed": True},
        {"id": "ms7", "title": "FastAPI / Express Server APIs", "completed": False},
        {"id": "ms8", "title": "Docker, Cloud Run & CI/CD", "completed": False}
      ]
    },
    {
      "id": "m3",
      "title": "Campus Placement Drill",
      "description": "Aptitude, Verbal Ability, Resume Polish & Mock Interviews",
      "milestones": [
        {"id": "ms9", "title": "Quantitative Aptitude (Speed, Distance, Ratios)", "completed": True},
        {"id": "ms10", "title": "STAR-formatted Resume Bullet Points", "completed": False},
        {"id": "ms11", "title": "3 Mock Tests Completed", "completed": False},
        {"id": "ms12", "title": "1-on-1 Senior Mentorship Review", "completed": False}
      ]
    }
  ],
  "Data Science & AI": [
    {
      "id": "ds_m1",
      "title": "Python & Mathematical Foundations",
      "description": "NumPy, Pandas, Linear Algebra & Probability",
      "milestones": [
        {"id": "ds_ms1", "title": "Python Data Analysis with Pandas & Seaborn", "completed": True},
        {"id": "ds_ms2", "title": "Descriptive & Inferential Statistics", "completed": False}
      ]
    },
    {
      "id": "ds_m2",
      "title": "Machine Learning & AI",
      "description": "Scikit-Learn, Regression, Classification & Gemini LLMs",
      "milestones": [
        {"id": "ds_ms3", "title": "Supervised Learning Models", "completed": False},
        {"id": "ds_ms4", "title": "LLM Prompting & Vector Search", "completed": False}
      ]
    }
  ],
  "Embedded & IoT Systems": [
    {
      "id": "emb_m1",
      "title": "Embedded C & Microcontrollers",
      "description": "C Programming, AVR/ARM Architecture & SPI/I2C Protocols",
      "milestones": [
        {"id": "emb_ms1", "title": "Pointers, Memory Allocation & Bit Manipulation", "completed": True},
        {"id": "emb_ms2", "title": "GPIO, Timers & Interrupt Handling", "completed": False}
      ]
    }
  ],
  "Cybersecurity": [
    {
      "id": "sec_m1",
      "title": "Network Security & Penetration Testing",
      "description": "TCP/IP, Wireshark, OWASP Top 10 & Cryptography",
      "milestones": [
        {"id": "sec_ms1", "title": "Network Packet Analysis with Wireshark", "completed": True},
        {"id": "sec_ms2", "title": "OWASP Top 10 Web Vulnerabilities", "completed": False}
      ]
    }
  ],
  "VLSI & Hardware Design": [
    {
      "id": "vlsi_m1",
      "title": "Digital Electronics & Verilog HDL",
      "description": "Combinational & Sequential Circuits, FSM & Verilog Simulation",
      "milestones": [
        {"id": "vlsi_ms1", "title": "Logic Gates & K-Map Minimization", "completed": True},
        {"id": "vlsi_ms2", "title": "Verilog Testbench Writing", "completed": False}
      ]
    }
  ]
}

File: frontend/src/data/mockData.ts
import {
  User,
  MockTest,
  InterviewQuestion,
  SeniorMentor,
  MentorshipPair,
  ResumeData,
  TestResult
} from '../types';

export { INITIAL_ROADMAPS } from './roadmaps';

export const DEMO_USERS: User[] = [
  {
    id: 'usr_mentee_1',
    name: 'Anand Nair',
    email: 'anand.nair@ucek.ac.in',
    role: 'mentee',
    year: '4th Year',
    branch: 'CSE',
    domain: 'Software Engineering',
    readinessScore: 78,
    avatar: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?auto=format&fit=crop&q=80&w=200'
  },
  {
    id: 'usr_admin_1',
    name: 'Dr. Suresh Kumar',
    email: 'placement.cell@ucek.ac.in',
    role: 'admin',
    year: 'Faculty',
    branch: 'Placement Cell',
    domain: 'Management & Consulting',
    readinessScore: 100,
    avatar: 'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?auto=format&fit=crop&q=80&w=200'
  }
];

export const MOCK_TESTS: MockTest[] = [];

export const INTERVIEW_QUESTIONS: InterviewQuestion[] = [
  {
    id: 'hr_1',
    questionText: 'Tell me about yourself and why you are interested in joining our organization as a Campus Recruit.',
    category: 'HR & Behavioral',
    difficulty: 'Easy',
    companyTag: 'TCS',
    suggestedAnswer: 'Start with your name, UCEK branch, core technical stack, key projects, internship experience, and align your personal growth with the company mission.'
  },
  {
    id: 'hr_2',
    questionText: 'Describe a challenging situation during your final year project where team conflict arose. How did you resolve it?',
    category: 'HR & Behavioral',
    difficulty: 'Medium',
    companyTag: 'TCS',
    suggestedAnswer: 'Use the STAR method (Situation, Task, Action, Result). Highlight active listening, objective data-driven compromise, and delivering on target.'
  },
  {
    id: 'inf_1',
    questionText: 'Why do you want to join Infosys as a Systems Engineer / Specialist Programmer, and where do you see yourself in 3 years?',
    category: 'HR & Behavioral',
    difficulty: 'Easy',
    companyTag: 'Infosys',
    suggestedAnswer: 'Discuss Infosys learning ecosystem (Mysore Training Center), continuous upskilling, and contribution to enterprise digital transformation.'
  },
  {
    id: 'inf_2',
    questionText: 'What is the difference between Process and Thread? Explain how Context Switching works.',
    category: 'Technical',
    difficulty: 'Medium',
    companyTag: 'Infosys',
    suggestedAnswer: 'A process is an isolated program in execution with its own memory space. A thread is a lightweight execution unit inside a process sharing memory space.'
  },
  {
    id: 'wip_1',
    questionText: 'How do you prioritize multiple tasks when working under tight deadlines in a multi-client project team?',
    category: 'Situational',
    difficulty: 'Medium',
    companyTag: 'Wipro',
    suggestedAnswer: 'Prioritize by business impact and urgency. Communicate early with team leads to clarify requirements and prevent bottlenecks.'
  },
  {
    id: 'wip_2',
    questionText: 'Describe a project where you had to learn a new technology or programming language quickly.',
    category: 'HR & Behavioral',
    difficulty: 'Easy',
    companyTag: 'Wipro',
    suggestedAnswer: 'Focus on your structured learning approach, documentation research, building a prototype, and delivering the module on schedule.'
  },
  {
    id: 'acc_1',
    questionText: 'Accenture focuses heavily on innovation and emerging tech. Share an example of how you used creative problem-solving in a project.',
    category: 'HR & Behavioral',
    difficulty: 'Hard',
    companyTag: 'Accenture',
    suggestedAnswer: 'Detail how you identified an operational flaw or bottleneck, researched innovative tools, and implemented an optimized automated solution.'
  },
  {
    id: 'acc_2',
    questionText: 'How do you handle constructive criticism or feedback from a senior developer or team mentor?',
    category: 'HR & Behavioral',
    difficulty: 'Easy',
    companyTag: 'Accenture',
    suggestedAnswer: 'Emphasize a growth mindset, taking feedback objectively, implementing changes promptly, and asking clarifying questions to improve code quality.'
  },
  {
    id: 'gen_1',
    questionText: 'What are your key strengths and what is one technical area you are actively working to improve?',
    category: 'HR & Behavioral',
    difficulty: 'Easy',
    companyTag: 'General HR',
    suggestedAnswer: 'Share 2 technical strengths backed by project work, and 1 area you are actively improving via certifications or daily coding practice.'
  },
  {
    id: 'gen_2',
    questionText: 'If you are given two high-priority tasks with tight deadlines simultaneously by two different project leads, how will you manage?',
    category: 'Situational',
    difficulty: 'Hard',
    companyTag: 'General HR',
    suggestedAnswer: 'Communicate transparently with both leads, assess business impact/dependencies, propose realistic split timelines, and escalate if blocking.'
  }
];

export const SENIOR_MENTORS: SeniorMentor[] = [
  {
    id: 'ment_1',
    name: 'Devika Suresh',
    avatar: 'https://images.unsplash.com/photo-1573496359142-b8d87734a5a2?auto=format&fit=crop&q=80&w=200',
    role: 'Software Engineer I',
    company: 'Google',
    domain: 'Software Engineering',
    bio: 'UCEK CSE Alumna (2025). Cracked Google Off-Campus & TCS Digital. Specializes in LeetCode algorithms, mock tech interviews, and ATS resume polish.',
    rating: 4.9,
    availability: '3 hrs / week'
  },
  {
    id: 'ment_2',
    name: 'Rahul Krishna',
    avatar: 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?auto=format&fit=crop&q=80&w=200',
    role: 'Embedded Systems Specialist',
    company: 'Texas Instruments',
    domain: 'Core Electronics & Embedded',
    bio: 'UCEK ECE Alumnus (2024). Placed at TI with focus on ARM Cortex microcontrollers, RTOS, & hardware prototyping.',
    rating: 4.8,
    availability: '2 hrs / week'
  },
  {
    id: 'ment_3',
    name: 'Ananya Pillai',
    avatar: 'https://images.unsplash.com/photo-1580489944761-15a19d654956?auto=format&fit=crop&q=80&w=200',
    role: 'Data Scientist',
    company: 'Amazon',
    domain: 'Data Science & AI',
    bio: 'UCEK IT Alumna (2024). Expert in ML pipelines, Python analytics, SQL case studies, and Amazon STAR behavioral interview rounds.',
    rating: 5.0,
    availability: '4 hrs / week'
  }
];

export const INITIAL_MENTORSHIP: MentorshipPair = {
  id: 'pair_101',
  mentorId: 'ment_1',
  mentorName: 'Devika Suresh',
  mentorCompany: 'Google',
  mentorRole: 'Software Engineer I',
  menteeId: 'usr_mentee_1',
  menteeName: 'Anand Nair',
  status: 'Active',
  nextMeetingDate: '2026-08-05 (Wed) @ 6:00 PM',
  logs: [
    {
      id: 'log_1',
      date: '2026-07-28',
      topic: 'Initial Diagnostic & Resume Review',
      feedback: 'Anand has solid foundational knowledge in Java & DSA. Resume needs quantified impact metrics in final year web app project.',
      actionItems: ['Quantify project metrics with STAR method', 'Solve 15 Sliding Window problems on LeetCode', 'Take TCS Mock Drive']
    }
  ]
};

export const INITIAL_RECENT_SCORES: TestResult[] = [];

export const INITIAL_RESUME_DATA: ResumeData = {
  template: 'ats',
  personal: {
    fullName: 'Anand Nair',
    email: 'anand.nair@ucek.ac.in',
    phone: '+91 98765 43210',
    location: 'Trivandrum, Kerala',
    linkedIn: 'linkedin.com/in/anandnair-ucek',
    github: 'github.com/anandnair-ucek',
    summary: 'Proactive 4th-year Computer Science Engineering student at UCEK with expertise in React, TypeScript, Node.js, and Data Structures. Passionate about building high-performance web applications and solving complex algorithmic challenges.'
  },
  education: [
    {
      id: 'edu_1',
      institution: 'University College of Engineering Kariavattom (UCEK)',
      degree: 'B.Tech',
      fieldOfStudy: 'Computer Science and Engineering',
      startDate: '2022',
      endDate: '2026',
      gpa: '8.4 / 10 CGPA'
    }
  ],
  experience: [
    {
      id: 'exp_1',
      company: 'Technopark Student Interns',
      position: 'Frontend Developer Intern',
      startDate: 'May 2025',
      endDate: 'July 2025',
      isCurrent: false,
      bullets: [
        'Developed responsive React UI components reducing load time by 35% across 4 primary modules.',
        'Integrated REST APIs with Axios and implemented Redux state management for 1,200 daily active users.'
      ]
    }
  ],
  projects: [
    {
      id: 'proj_1',
      title: 'Smart Campus Placement Portal',
      techStack: 'React, TypeScript, Tailwind CSS, Node.js',
      description: 'Built a centralized campus recruitment platform featuring automated ATS resume parsing and timed mock test drives.',
      link: 'github.com/anandnair-ucek/placement-portal',
      bullets: [
        'Architected real-time quiz engine supporting 500+ concurrent student exam submissions.',
        'Engineered AI resume review system evaluating keyword match accuracy with 92% precision.'
      ]
    }
  ],
  skills: [
    { id: 'sk_1', category: 'Programming Languages', items: 'JavaScript, TypeScript, Java, C++, Python, SQL' },
    { id: 'sk_2', category: 'Web Frameworks & Libraries', items: 'React.js, Node.js, Express, HTML5, CSS3, Tailwind CSS' },
    { id: 'sk_3', category: 'Tools & Databases', items: 'Git, GitHub, VS Code, Postman, MySQL, MongoDB' }
  ],
  certifications: [
    'NPTEL Online Certification: Data Structures & Algorithms in Java (Elite Badge)',
    'Meta Front-End Developer Professional Certificate (Coursera)'
  ]
};
